src/sentry/sentry_metrics/querying/data_v2/transformation.py

```diff
@@ -180,6 +184,10 @@ def transform(self) -> Mapping[str, Any]:
         """
         Transforms the query results into the Sentry's API format.
         """
+        # If we have not run any queries, we won't return anything back.
+        if not self._query_results:
```

Comment:
Just added this to handle possible edge cases in which the executor returns empty results but this should not happen.

Reply:
maybe we could emit a sentry error/message/metric here? If it never happens I'd rather not have it, just adds noise 😅